modules/gapi/test/common/gapi_render_tests.hpp

```diff
@@ -0,0 +1,70 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+//
+// Copyright (C) 2018 Intel Corporation
+
+
+#ifndef OPENCV_GAPI_RENDER_TESTS_HPP
+#define OPENCV_GAPI_RENDER_TESTS_HPP
+
+#include "gapi_tests_common.hpp"
+#include "api/render_priv.hpp"
+
+namespace opencv_test
+{
+
+using Points            = std::vector<cv::Point>;
+using Rects             = std::vector<cv::Rect>;
+using PairOfPoints      = std::pair<cv::Point, cv::Point>;
+using VecOfPairOfPoints = std::vector<PairOfPoints>;
+
+template<class T>
+class RenderTestParams : public TestParams<T>
+{
+protected:
+    void Init(const cv::Size& size, bool isNV12)
+    {
+        MatType type = CV_8UC3;
+        bool initOut = true;
+
+        TestFunctional::initMatsRandU(type, size, type, initOut);
+
+        if (isNV12) {
+            cv::gapi::wip::draw::BGR2NV12(TestFunctional::out_mat_ocv, y, uv);
+            cv::cvtColorTwoPlane(y, uv, TestFunctional::out_mat_ocv, cv::COLOR_YUV2BGR_NV12);
+        }
+    }
+
+    void Run(cv::Mat& out_mat, cv::Mat& ref_mat, bool isNV12)
+    {
+        if (isNV12) {
+            cv::gapi::wip::draw::BGR2NV12(out_mat, y, uv);
+            cv::gapi::wip::draw::render(y, uv, prims);
+            cv::cvtColorTwoPlane(y, uv, out_mat, cv::COLOR_YUV2BGR_NV12);
+
+            cv::gapi::wip::draw::BGR2NV12(ref_mat, y, uv);
```

Comment:
I'd comment that this is done due to data loss when converting to NV12 and back, that's why `ref_mat` undergoes the same transformation

Reply:
You're right, added

Reply:
consider this resolved. for some reason I cannot resolve my discussions so feel free to resolve yourself